### PR TITLE
[12.0] split beesdoo_easy_my_coop

### LINKED ISSUE #276: Rename `beesdoo_easy_my_coop`
- [ ] ?

### LINKED ISSUE #337: Split `beesdoo_emc` into `cooperator_eater_configuration`

### LINKED ISSUE #338: Split `beesdoo_emc` into `cooperator_worker_configuration`

### LINKED ISSUE #339: Split `beesdoo_emc` into `cooperator_info_session`

### PR DESCRIPTION
**Pull request by victor-champonnois**
## Description
Split beesdoo_easy_my_coop into:
- cooperator_eater rather then cooperator_eater_configuration
- cooperator_worker rather then cooperator_worker_configuration
- cooperator_info_session

Fix https://github.com/beescoop/Obeesdoo/issues/337
Fix https://github.com/beescoop/Obeesdoo/issues/276
Fix https://github.com/beescoop/Obeesdoo/issues/338
Fix https://github.com/beescoop/Obeesdoo/issues/339

**Misc**
- removed `test_max_eater_assignment_share_b`, I don't see a meaningful difference with `test_max_eater_assignment_share_a`
- removed `test_multiple_eater_assignement_share_a` and `test_parent_assignement_to_eater`: this is testing the ORM, not our code

---

### deprecated notes

Some questions I have:
- I put the `cooperator_type` field that was originally in `beesdoo_base` in `cooperator_worker_configuration`. I don't really understand the point of this field, but this seemed like a good place. @robinkeunen do you have an opinion ?
- `beesdoo_easy_my_coop` had an `auto-install` parameter, which means it installed itself automatically when its dependencies ("beesdoo_base", "beesdoo_shift", "cooperator", "cooperator_website", "partner_contact_birthdate") were installed. Should we keep this behavior for the splitted modules ? And if yes, for all of them ? (@vdewulf maybe you have an answer to that?) 

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

**Comment by vdewulf** [comment]
@victor-champonnois About that:
`beesdoo_easy_my_coop had an auto-install parameter, which means it installed itself automatically when its dependencies ("beesdoo_base", "beesdoo_shift", "cooperator", "cooperator_website", "partner_contact_birthdate") were installed. Should we keep this behavior for the splitted modules ? And if yes, for all of them ? (@vdewulf maybe you have an answer to that?)`

I don't know what you will keep in the new splitted module, so I don't know if it would make sense to keep this behavior or not. Let's talk?

**Comment by vdewulf** [comment]
Answer to beesdoo_easy_my_coop had an auto-install parameter, which means it installed itself automatically when its dependencies ("beesdoo_base", "beesdoo_shift", "cooperator", "cooperator_website", "partner_contact_birthdate") were installed. Should we keep this behavior for the splitted modules ? And if yes, for all of them ? (@vdewulf maybe you have an answer to that?)
Decision vdewulf/vchamponnois ===> no autoinstall

**Comment by victor-champonnois** [comment]
status: waiting for beesdoo_base to deploy in test

**Comment by robinkeunen** [comment]
I added a help message following discussion with @ferdymercury https://github.com/beescoop/Obeesdoo/issues/439#issuecomment-1196477827

**Comment by robinkeunen** [comment]
@victor-champonnois @remytms @huguesdk j'ai mis à jour la description de la tâche, prêt pour une revue.

File: cooperator_worker/models/res_partner.py
Comment (victor-champonnois) [inline cooperator_worker/models/res_partner.py]: I have difficulty understanding the help message. Is it possible to make it clearer?

**Review by victor-champonnois — COMMENTED** [review]
LGTM after a quick review, but I cannot approve because I created the PR ^^

**Comment by robinkeunen** [comment]
@polchampion it's been deployed on bees-test, lenid-test, lepedalo-test and spp-test. Can you have a look ? Here are the descriptions of the new modules :

```
# cooperator_worker
- Add working and shopping information on a Share product
- Overwrite is_worker and can_shop based on share configuration.

# cooperator_eater
- Overwrite validation of subscription request to get eaters information
- Add constraint on eaters based on share configuration
- Add eater configuration on share product

# cooperator_info_session
Configuration for info session on website form for getting a cooperator share :

- info session description and fields on the online form
- add info session text description on the company
- set if the attending an info session is required on the company
```

This will need a release note and probably an update of the documentation.

File: cooperator_worker/models/res_partner.py
Comment (robinkeunen) [reply cooperator_worker/models/res_partner.py]: done

File: cooperator_worker/README.rst
Comment (polchampion) [inline cooperator_worker/README.rst]: @victor-champonnois Will the strings "easy my coop" in readme be changed later ? Otherwise, best to update the readme now. (see ligne 22 as well)

File: cooperator_worker/i18n/cooperator_worker_configuration.pot
Comment (polchampion) [inline cooperator_worker/i18n/cooperator_worker_configuration.pot]: @victor-champonnois @robinkeunen Je ne sais pas si les parts A, B et C de la bees sont toujours hardcodées ailleurs dans le code, mais elles sont ici dans le pot. Est-ce maintenant l'occasion de se débarrasser de ceci ?

**Comment by polchampion** [comment]
> # cooperator_worker
> - Overwrite is_worker and can_shop based on share configuration.

@victor-champonnois How is `can_shop` overwritten and based on what ? I thought there were no links between Cooperators and `can_shop`. Is `can_shop` introduced in `beesdoo_shift` or `cooperator` or both ?

> # cooperator_eater
> - Overwrite validation of subscription request to get eaters information

@victor-champonnois I don't undestand what this means

File: .github/workflows/test.yml
Comment (robinkeunen) [inline .github/workflows/test.yml:35]: Est-ce que ceci est auto-généré depuis copier ? Sinon, on peut juste retirer les tests de polln et macavrac

File: cooperator_worker/README.rst
Comment (robinkeunen) [reply cooperator_worker/README.rst]: merci 👍

File: cooperator_worker/i18n/cooperator_worker_configuration.pot
Comment (robinkeunen) [reply cooperator_worker/i18n/cooperator_worker_configuration.pot]: Ce sont les traductions de données de démo. J'ai mis à jour les traductions, peux-tu y jeter un oeil ?

**Comment by robinkeunen** [comment]
@polchampion fixed the module descriptions.

File: .github/workflows/test.yml
Comment (huguesdk) [reply .github/workflows/test.yml:35]: yes, copier asks for rebel module groups. they are tested separately. i don’t know whether it is possible to simply skip some modules. i’d rather have modules tested separately instead of not at all, because that would mean that we wouldn’t be aware if they got broken by some commit.

**Comment by polchampion** [comment]
Tested on `lepedalo-test`

> # cooperator_worker
- [x] Working and shopping information appear on a Share product
- [x] `is_worker` is checked based on share configuration
- [x] `can_shop` is checked based on share config unless overwritten by worker status.

> # cooperator_eater
- [x] Overwrite validation of subscription request to get eaters information  --> I don't understand what this means @robinkeunen 
- [x] Add constraint on eaters based on share configuration : Ok for effective cooperators
NO CONSTRAINT on eaters based on share configuration for candidate cooperators (as before) --> should the "eater" tab appear on candidate cooperators ?
- [x] Eater configuration appears on share product

> # cooperator_info_session
- [x] Configuration for info session appears on the company config (activate + required?)
- [x] Info session description and checkbox appears on website form

File: .github/workflows/test.yml
Comment (robinkeunen) [reply .github/workflows/test.yml:35]: I think we can exclude them by removing the configuration from the test.yaml: there are no tests in those modules and we do not maintain them.

=> less time & less energy.

**Comment by robinkeunen** [comment]
@polchampion DESCRIPRION.rst was changed to 

> - Set new cooperator as eater or worker based on the share configuration

File: cooperator_worker/i18n/fr.po
Comment (polchampion) [inline cooperator_worker/i18n/fr.po]: @robinkeunen  Ma suggestion pour la traduction: "Ajouter le candidat coopérateur au système des shifts". La fonctionnalité n'est pas affichée actuellement, cf [cette issue](https://github.com/beescoop/Obeesdoo/issues/374)

File: cooperator_worker/i18n/cooperator_worker_configuration.pot
Comment (polchampion) [reply cooperator_worker/i18n/cooperator_worker_configuration.pot]: All good except one comment (see reply)

**Review by robinkeunen — APPROVED** [review]

**Pull request merged**
